### PR TITLE
Show email if the authenticated user owns the profile page being requested for

### PR DESCRIPTION
**Pull request by adelowo**
Also removed `setting.UI.ShowUserEmail` as it's documentation says it
only controls the email setting on the explore page 

> Would fix broken CI, this was pushed in a quick rush

### CODE REVIEW

File: routers/user/profile.go [hunk 1]
```diff
@@ -233,7 +233,8 @@ func Profile(ctx *context.Context) {
 		}
 	}
 
-	ctx.Data["ShowUserEmail"] = setting.UI.ShowUserEmail
+	ctx.Data["ShowUserEmail"] = (len(ctxUser.Email) > 0 && ctx.IsSigned && !ctxUser.KeepEmailPrivate) ||
```
Comment (lafriks) [inline routers/user/profile.go]: Condition should probably be:
```
len(ctxUser.Email) > 0 && ctx.IsSigned 
   && (!ctxUser.KeepEmailPrivate || ctxUser.ID == ctx.User.ID)
```

Comment (adelowo) [reply routers/user/profile.go]: I hope to get some free time tomorrow and test locally ( fix ) too

Comment (lafriks) [reply routers/user/profile.go]: My provided condition should work also for unauthorized users

**Comment by adelowo** [comment]
~~Bummer, will update this PR~~

Will sure do this over the new week

**Review by lafriks — APPROVED** [review]

**Review by techknowlogick — APPROVED** [review]

**Pull request merged**